src/unix/stream.c

```diff
@@ -1003,8 +1003,10 @@ static int uv__stream_recv_cmsg(uv_stream_t* stream, struct msghdr* msg) {
         err = uv__stream_queue_fd(stream, fd);
         if (err != 0) {
           /* Close rest */
-          for (; i < count; i++)
+          for (; i < count; i++) {
+            memcpy(&fd, (char*) CMSG_DATA(cmsg) + i * sizeof(fd), sizeof(fd));
             uv__close(fd);
+          }
           return err;
```

Comment:
Hm, that's a rather blatant bug now that you point it out. Leaking file descriptors _and_ double-closing the wrong file descriptor.

Looking at this function more closely, returning here is also wrong because that means subsequent control messages are ignored (the outer loop), potentially resulting in more file descriptor leaks.

This function is in need of a more drastic rewrite, see #4358.